perfkitbenchmarker/publisher.py

```diff
@@ -487,8 +487,11 @@ class SampleCollector(object):
       SampleCollector._DefaultPublishers.
     run_uri: A unique tag for the run.
   """
-  def __init__(self, metadata_providers=None, publishers=None):
-    self.samples = []
+  def __init__(self, samples=None, metadata_providers=None, publishers=None):
+    if samples is None:
```

Comment:
The very specific reason is because we used to use RunThreaded for benchmarks, passing them all the same SampleCollector and expecting all samples to be aggregated in its samples list. Now that we're using RunParallelProcesses, the samples list in the SampleCollector received by each process needs to refer to the same list. So instead of relying on the default serializing/deserializing behavior, which would give each process a different list (even though originally only a single SampleCollector is passed in), I give the SampleCollector a proxy to a list managed by a SyncManager. That way, the deserialized SampleCollector in every process will append to the same list. Take a look at the modified RunBenchmarks in pkb.py to see how the managers are used.